cmd/observe/observe.go

```diff
@@ -0,0 +1,192 @@
+ *                |  ██║  ██║╚██████╔╝██║  ██║██║  ██║███████╗ |              *
+ *                |  ╚═╝  ╚═╝ ╚═════╝ ╚═╝  ╚═╝╚═╝  ╚═╝╚══════╝ |              *
+ *                +--------------------------------------------+              *
+ *                                                                            *
+ *                         Distributed Systems Runtime                        *
+ *                                                                            *
+ * -------------------------------------------------------------------------- *
+ *                                                                            *
+ *   Licensed under the Apache License, Version 2.0 (the "License");          *
+ *   you may not use this file except in compliance with the License.         *
+ *   You may obtain a copy of the License at                                  *
+ *                                                                            *
+ *       http://www.apache.org/licenses/LICENSE-2.0                           *
+ *                                                                            *
+ *   Unless required by applicable law or agreed to in writing, software      *
+ *   distributed under the License is distributed on an "AS IS" BASIS,        *
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. *
+ *   See the License for the specific language governing permissions and      *
+ *   limitations under the License.                                           *
+ *                                                                            *
+\* -------------------------------------------------------------------------- */
+
+package observe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net"
+
+	"github.com/aurae-runtime/ae/pkg/cli"
+	"github.com/aurae-runtime/ae/pkg/cli/printer"
+	"github.com/aurae-runtime/ae/pkg/client"
+	"github.com/aurae-runtime/ae/pkg/config"
+	"github.com/spf13/cobra"
+
+	aeCMD "github.com/aurae-runtime/ae/cmd"
+	observev0 "github.com/aurae-runtime/ae/pkg/api/v0/observe"
+)
+
+type outputObserve struct{}
+
+type option struct {
+	aeCMD.Option
+	ctx          context.Context
+	ip           string
+	logtype      string
+	port         uint16
+	protocol     string
+	verbose      bool
+	writer       io.Writer
+	output       *outputObserve
+	outputFormat *cli.OutputFormat
+}
+
+func (o *option) Complete(args []string) error {
+	log.Println(args)
+	if len(args) != 2 {
+		return errors.New("expected ip address and log type to be passed to this command")
+	}
+	o.ip = args[0]
+	o.logtype = args[1]
+	return nil
+}
+
+func (o *option) Validate() error {
+	if err := o.outputFormat.Validate(); err != nil {
+		return err
+	}
+	if len(o.ip) == 0 {
+		return errors.New("ip address must be passed to this command")
+	}
+	if ip := net.ParseIP(o.ip); ip == nil {
+		return fmt.Errorf("failed to parse IP %q", o.ip)
+	}
```

Comment:
I wonder: does it make sense to test for a valid port aswell?

> Valid values range from 1 through 65535. **However, some of the ports in the range 1 through 1023 are used by system-supplied TCP/IP applications.** If the user specifies one of these ports, it can affect the operation of those applications.

Reply:
the flag itself is defined as a uint16 so i don't think it can be invalid.  i also don't want to make assumptions about what ports people might choose for auraed when it's running as PID1.